### PR TITLE
knative-kubecon: adding build part

### PR DESCRIPTION
**Pull request by vdemeester**
… standalone build-part, with a `serving` example that should work :angel:

cc @pmorie @markusthoemmes

Next step is to decide which "source" we use in common (for all part of the demo) and move these at the right place.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by markusthoemmes** [comment]
Neat, thanks for adding in the serving part straight away!

File: knative-kubecon/build/020-serving.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          - name: IMAGE
+            value: "helloworld:latest"
+          - name: NAME
+            value: foo-build
+      revisionTemplate:
+        metadata:
+          annotations:
+            alpha.image.policy.openshift.io/resolve-names: "*"
+        spec:
+          container:
```
Comment (markusthoemmes) [inline knative-kubecon/build/020-serving.yaml:29]: can we add `containerConcurrency: 1` above, so we "force" autoscaling to work a bit harder (makes for an easier demo)

Comment (vdemeester) [reply knative-kubecon/build/020-serving.yaml:29]: updated :wink:

File: knative-kubecon/build/010-build.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apiVersion: build.knative.dev/v1alpha1
+kind: Build
+metadata:
+  name: oc-build-1
+spec:
+  source:
+    git:
+      url: https://github.com/vdemeester/os-sample-python
```
Comment (matzew) [inline knative-kubecon/build/010-build.yaml:8]: the idea is to hook that code also to eventing, using a `ksvc`, right @vdemeester @markusthoemmes  ?

Comment (vdemeester) [reply knative-kubecon/build/010-build.yaml:8]: @matzew in the end, I think so yeah :angel: this PR "just" import the raw build in the repo, to be mangled, updated to the final demo script/workflow/data :angel:

Comment (markusthoemmes) [reply knative-kubecon/build/010-build.yaml:8]: @matzew correct, we're creating a service already in the 020-serving.yaml to be connected to eventing.

File: knative-kubecon/build/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# […]
+$ oc get pods
+$ oc get build
+$ oc get buildconfig
+# Clean the build (for now)
+$ oc delete -f 010-build.yaml
+# Create a serving that will depend on a build (creating this one too)
+# update the knative config map to add the internal registry as "safe"
+# to not resolve.
+$ val=$(oc -n knative-serving get cm config-controller -oyaml | yq r - data.registriesSkippingTagResolving | awk '{print $1",docker-registry.default.svc:5000"}')
```
Comment (pmorie) [inline knative-kubecon/build/README.md]: I needed to change this command to get it to work:

```
val=$(oc -n knative-serving get cm config-controller -o yaml | yq -r .data.registriesSkippingTagResolving | awk '{print $1",docker-registry.default.svc:5000"}')
```

File: knative-kubecon/build/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+$ oc -n knative-serving get cm config-controller -oyaml | yq w - data.registriesSkippingTagResolving $val | oc apply -f -
```
Comment (pmorie) [inline knative-kubecon/build/README.md:23]: I wasn't able to get this to work on my machine :-/ and had to do it manually.

File: knative-kubecon/build/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# to not resolve.
+$ val=$(oc -n knative-serving get cm config-controller -oyaml | yq r - data.registriesSkippingTagResolving | awk '{print $1",docker-registry.default.svc:5000"}')
+$ oc -n knative-serving get cm config-controller -oyaml | yq w - data.registriesSkippingTagResolving $val | oc apply -f - 
+$ oc apply -f 020-serving.yaml
+# […]
+$ oc get pods
+# […] builds and pod should be created and running, …
+# Once the build is done and service running
+$ curl -H "Host: helloworld-openshift-python.myproject.example.com" http://$(minishift ip):32380
+Hello Openshift!
```
Comment (pmorie) [inline knative-kubecon/build/README.md]: This printed "Hello Knative & OpenShift" for me :)

**Review by pmorie — COMMENTED** [review]
I had some difficulty with the yq commands. I wonder if there are different binaries that support different sets of arguments.

File: knative-kubecon/build/README.md (hunk 5 above)
Comment (vdemeester) [reply knative-kubecon/build/README.md]: Oh right, I didn't update that line :sweat:

**Comment by pmorie** [comment]
I would be fine merging this as-is and tuning the commands in a follow-up.

File: knative-kubecon/build/README.md (hunk 4 above)
Comment (vdemeester) [reply knative-kubecon/build/README.md:23]: I'll try to make it work in a follow-up :wink:

**Pull request merged**